internal/webhooks/cluster.go

```diff
@@ -395,6 +390,227 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 	return allWarnings, allErrs
 }
 
+func (webhook *Cluster) validateTopologyVersion(ctx context.Context, fldPath *field.Path, fldValue string, inVersion, oldVersion semver.Version, oldCluster *clusterv1.Cluster) field.ErrorList {
+	// Version could only be increased.
+	if inVersion.NE(semver.Version{}) && oldVersion.NE(semver.Version{}) && version.Compare(inVersion, oldVersion, version.WithBuildTags()) == -1 {
+		return field.ErrorList{field.Invalid(
+			fldPath,
+			fldValue,
+			fmt.Sprintf("version cannot be decreased from %q to %q", oldVersion, inVersion),
+		)}
+	}
+
+	// A +2 minor version upgrade is not allowed.
+	ceilVersion := semver.Version{
+		Major: oldVersion.Major,
+		Minor: oldVersion.Minor + 2,
+		Patch: 0,
+	}
+	if inVersion.GTE(ceilVersion) {
+		return field.ErrorList{field.Invalid(
+			fldPath,
+			fldValue,
+			fmt.Sprintf("version cannot be increased from %q to %q", oldVersion, inVersion),
+		),
+		}
+	}
+
+	// Only check the following cases if the minor version increases by 1 (we already return above for >= 2).
+	ceilVersion = semver.Version{
+		Major: oldVersion.Major,
+		Minor: oldVersion.Minor + 1,
+		Patch: 0,
+	}
+
+	// Return early if its not a minor version upgrade.
+	if !inVersion.GTE(ceilVersion) {
+		return nil
+	}
+
+	allErrs := field.ErrorList{}
+	// minor version cannot be increased if control plane is upgrading or not yet on the current version
+	if err := validateTopologyControlPlaneVersion(ctx, webhook.Client, oldCluster, oldVersion); err != nil {
+		allErrs = append(allErrs, field.Invalid(
+			fldPath,
+			fldValue,
+			fmt.Sprintf("blocking version update due to ControlPlane version check: %v", err),
```

Comment:
Instead of creating many field errors for the same field (one for CP, MD, MP), I think we should keep track of the different errors, and then create a single field error for version down at line 459 (you can prefix the list of errors with something like "minor version update cannot happen at this time: <list off errors>")

Reply:
Output now with two simplified errors ("foo" and "bar") (test-case):

```
                Cluster.cluster.x-k8s.io "cluster1" is invalid: spec.topology.version: Invalid value: "v1.20.2": minor version update cannot happen at this time: [foo, bar]
                {
                    ErrStatus: {
                        TypeMeta: {Kind: "", APIVersion: ""},
                        ListMeta: {
                            SelfLink: "",
                            ResourceVersion: "",
                            Continue: "",
                            RemainingItemCount: nil,
                        },
                        Status: "Failure",
                        Message: "Cluster.cluster.x-k8s.io \"cluster1\" is invalid: spec.topology.version: Invalid value: \"v1.20.2\": minor version update cannot happen at this time: [foo, bar]",
                        Reason: "Invalid",
                        Details: {
                            Name: "cluster1",
                            Group: "cluster.x-k8s.io",
                            Kind: "Cluster",
                            UID: "",
                            Causes: [
                                {
                                    Type: "FieldValueInvalid",
                                    Message: "Invalid value: \"v1.20.2\": minor version update cannot happen at this time: [foo, bar]",
                                    Field: "spec.topology.version",
                                },
                            ],
                            RetryAfterSeconds: 0,
                        },
                        Code: 422,
                    },
                }
```